carbon-design-system/carbon-addons-iot-react
# feat(dashboardeditor): support editable title widgets natively

Closes #

**Summary**

- feat(dashboardeditor): currently all the edit state behavior has to be kept in the parent widget.  This moves the state for the edit title capability into the dashboardeditor
- Improve the PageTitleBar to allow custom titles to be rendererd

**Change List (commits, features, bugs, etc)**

- feat(DashboardEditorHeader): add state tracking and widgets to edit the dashboard title
- feat(PageTitleBar): allow passing of a customer renderTitleFunction so alternate nodes can be returned than the default title string

**Acceptance Test (how to verify the PR)**

- Verify the custom render function in the pagetitlebar stories
- Test DashboardEditor story for editing the dashboard title

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Verify that normal pagetitlebar title is non-editable, verify if a dashboardeditor doesn't have a onEditTitle handler that it doesn't render the edit button
